server/src/main/java/org/opensearch/common/remote/RemoteWriteableEntity.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.remote;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * An interface to which provides defines the serialization/deserialization methods for objects to be uploaded to or downloaded from remote store.
+ * This interface is agnostic of the remote storage type.
+ *
+ * @param <T> The object type which can be uploaded to or downloaded from remote storage.
+ */
+public interface RemoteWriteableEntity<T> {
+
+    /**
+     * @param object The object T which is to be set in this writable entity
+     */
+    public void set(T object);
+
```

Comment:
Why can't ctor be used for this?

Reply:
When the `RemoteBlobStore` reads `RemoteWritableEntity`, the `RemoteWritableEntity` contains only the blob name. After reading the entity T from remote store, it uses this method to set the entity T and return.